host_core/native/hostcore_wasmcloud_native/src/wasmruntime.rs

```diff
@@ -1,6 +1,9 @@
-use anyhow::{self, Context};
+use anyhow::{self, bail, Context};
 use async_trait::async_trait;
-use log::error;
+use rand::{thread_rng, Rng, RngCore};
+use tracing::{error, trace};
```

Comment:
we can't use the tracer that requires a subscriber. we basically need to use "regular" rust logging via one of the other crates.

Reply:
What's the difference?

Reply:
the tracing subscriber manages its own threads/async, which can and will cause a NIF to fail. Without explicitly putting things on dirty IO or CPU schedulers, the NIF will shut down any background thread that lasted more than some number of microseconds. In other words, if we spin up a tracing subscriber during the resource initialization without NIF-specific stuff around it, that thread is likely to be destroyed. 

We need the most basic of logging implementations here that don't involve async, e.g. something like simplelog etc

Reply:
That means we will lose all the instrumentation provided by `wasmcloud` library and e.g. `wasmtime` that could be very useful for things like debugging and profiling (by generating a flame graph)

What kind of "NIF-specific stuff" would we need to facilitate this?

Couldn't we just flush before returning control back to Erlang?

Reply:
I haven't spiked this, but my guess is we can write a tracing subscriber that we spawned using a NIF-friendly spawn, e.g. https://docs.rs/rustler/latest/rustler/thread/fn.spawn.html